### ISSUE TITLE
FR: Add option to specify interval to plugin.RegisterRead.

### ISSUE THREAD

**Issue by iMilnb** [collectd/go-collectd#35]
I'm writing a simple plugin that needs to be run less often than every 10s, in order to achieve this I have the following values:

```
	vl := api.ValueList{
		Identifier: api.Identifier{
			Host:         exec.Hostname(),
			Plugin:       "foo",
			Type:         "gauge",
			TypeInstance: "ticker",
		},
		Time: time.Now(),
		Interval: 60 * time.Second,
		DSNames:  pairs,
		Values:   vals,
	}
```
Nevertheless, the plugin runs every 10 seconds.

I also tried overriding using the `<LoadPlugin>` method in `collectd.conf` without success.

Am I hitting a bug or did I miss something?

Thanks

**Comment by octo** [comment]
Hi @iMilnb,

The `ValueList.Interval` field tells collectd how often to expect metrics, but does not control how often your coffee is run. How do you trigger these statements?

—octo

**Comment by iMilnb** [comment]
Hi @octo thanks for answering :)

The code is pretty much a copy of the example plugin, it is built as explained in the documentation:

```
$ go build -buildmode=c-shared -o bitstamp.so bitstamp.go
```

Copied into `/usr/lib/collectd/`, and `collectd.conf` has:

```
LoadPlugin bitstamp
```

I also tried using:

```
<LoadPlugin bitstamp>
        Interval 60
</LoadPlugin>
```
But it has no effect.

With the `python` module, the ReadRegister call looks like this `collectd.register_read(readvals, interval))` so it's possible to pass the interval how the plugin is called.

Any idea how to achieve this? Some websites APIs do not like being called that often...

**Comment by iMilnb** [comment]
FWIW I can modify plugin's call frequency by modifying the global "Interval" parameter in `collectd.conf`.

**Comment by octo** [comment]
This feature request would be solved by PR #43.
